File: wordcount-api/src/main/java/io/pivotal/demo/functions/WindowedWordCountFunction.java
package io.pivotal.demo.functions;

import java.util.List;

import org.springframework.data.gemfire.function.annotation.FunctionId;
import org.springframework.data.gemfire.function.annotation.OnRegion;

@OnRegion(region = "Words_Count_Windowed")
public interface WindowedWordCountFunction {

	@SuppressWarnings("rawtypes")
	@FunctionId("ComputeWindowedWordCount")
	List computeWindowedWordCount(Long startTimeInMillis, Long endTimeInMillis);
}

File: web-ui/src/main/kotlin/io/pivotal/demo/functions/WindowedWordCountFunction.kt
package io.pivotal.demo.functions

import io.pivotal.demo.domain.WordCount
import io.pivotal.demo.domain.WordCountWindowed
import org.springframework.data.gemfire.function.annotation.FunctionId
import org.springframework.data.gemfire.function.annotation.OnRegion

@OnRegion(region = "Words_Count_Windowed")
interface WindowedWordCountFunction {

    @FunctionId("ComputeWindowedWordCount")
    fun computeWindowedWordCount(startTimeInMillis: Long?, endTimeInMillis: Long?): HashMap<String, Long>
}
